### PR TITLE
[Feature] Update skill browser dialog

### LINKED ISSUE #7394: ✨ Modify behaviour in the skill picker dialog
image

## ✨ Summary

- [Figma file (form)](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=97-30604&mode=design&t=bN497K7OAcaoquHs-4)
- [Figma file (dialog)](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=97-31278&mode=design&t=bN497K7OAcaoquHs-4)

Lauren has requested that we adjust the skill browser dialog interaction to force users to select a skill family before continuing. This is being done for a few reasons:
- To make sure users know that skill families exist and how they are structured
- To avoid the frustration of searching the full list and coming up with no results
- To avoid users scrolling through the full list of skills by default

```[tasklist]
### Requirements
- [x] Update the dialog title, subtitle, and actions to match the latest designs
- [x] Restructure the form to ensure users are forced to select a family before presenting the skill search bar and remaining fields
- [x] Ensure all field labels are updated to match the design
- [x] Update the skill family filter to show new options and skill counts (`All skills`, `My library`, etc.) and ensure that `Programming languages and tools` is last in the list
- [x] Ensure that the new `My library` option in the skill family filter is optional so that we can show it in specific contexts
- [x] Add new null states to account for each step
- [x] Move the `help` accordion to the bottom of the form and collapse it by default
- [x] Ensure all elements are appropriately hidden at each step based on the new designs
- [x] Ensure that AT users understand new information has been loaded when the form's state changes
```

## 📸 Screenshots

<details><summary>Skill family filter (Default)</summary>

</details> 

<details><summary>Skill family filter (With My library)</summary>

</details> 

<details><summary>Initial state</summary>

</details> 

<details><summary>Family selected</summary>

</details> 

<details><summary>Skill selected</summary>

</details>

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #7394 

## 👋 Introduction

Updates the skill browser dialog copy, interactions and order.

## 🕵️ Details

- Updates most copy within the dialog
- Hides skill selection until a family has been selected
- Allows adding existing skills to the family option, labelled as "My library"
- Moved help accordion to bottom

### 🏷️ Labels

The designs labels were instructional which should be avoided, if possible. A label should describe the purpose of the input not provide instructions. Due to this, the labels were slightly tweaked to remove the instructions from them.

 - https://github.com/GCTC-NTGC/gc-digital-talent/issues/4805

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Run storybook `npm run storybook`
2. Navigate to the "Skill dialog" stories
3. Interact with them confirming they match [the mockup and AC](https://github.com/GCTC-NTGC/gc-digital-talent/issues/7394)

## 📸 Screenshot

Screenshot 2023-07-31 090653

### CODE REVIEW

File: apps/web/src/components/SkillDialog/SkillDialog.tsx [hunk 1]
```diff
@@ -6,35 +6,36 @@ import PlusCircleIcon from "@heroicons/react/20/solid/PlusCircleIcon";
...
 const defaultFormValues: FormValues = {
-  category: "all",
-  family: "all",
+  category: "",
+  family: "",
   skill: "",
 };
 
 interface SkillDialogProps {
+  // Al;l available skills
```
Comment (vd1992) [inline apps/web/src/components/SkillDialog/SkillDialog.tsx:23]: Typo

**Comment by vd1992** [comment]
An aside, shouldn't this add up to 185 and not 184 🤔 

image

**Comment by esizer** [comment]
> An aside, shouldn't this add up to 185 and not 184 🤔

Not necessarily, skills can exist in multiple categories.

File: apps/web/src/components/SkillDialog/SkillDialog.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/components/SkillDialog/SkillDialog.tsx]: Oops 🤭 Fixed in [fix typo](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7445/commits/1370e4bcf10a3d6a7664657edd6b83d61b4c994a)

File: apps/web/src/components/SkillDialog/SkillDialog.tsx
Comment (vd1992) [inline apps/web/src/components/SkillDialog/SkillDialog.tsx:1]: Acceptance criteria mismatch I believe

image

With respect to

image

Comment (vd1992) [inline apps/web/src/components/SkillDialog/SkillDialog.tsx:1]: Another thing, null state to alert user to select a skill family first

image

WRT

image

File: apps/web/src/components/SkillDialog/SkillDialog.stories.tsx [hunk 2]
```diff
@@ -42,3 +45,16 @@ LibraryContext.args = {
...
   showCategory: false,
 };
+
+export const ShowcaseContext = Template.bind({});
+ShowcaseContext.args = {
+  context: "showcase",
+  showCategory: false,
+};
+
+export const ShowcaseShowMyLibraryContext = Template.bind({});
```
Comment (vd1992) [inline apps/web/src/components/SkillDialog/SkillDialog.stories.tsx:55]: I think I am missing something, both new stories look the same

image

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
A few things to bring to your attention

File: apps/web/src/components/SkillDialog/SkillDialog.stories.tsx (hunk 2 above)
Comment (esizer) [reply apps/web/src/components/SkillDialog/SkillDialog.stories.tsx:55]: They are pretty darn close but the last one includes a family labelled "My library" which is just those you have added to your skills library (provided by the prop `inLibrary`)

File: apps/web/src/components/SkillDialog/SkillDialog.tsx
Comment (esizer) [reply apps/web/src/components/SkillDialog/SkillDialog.tsx:1]: Couldn't move to bottom for accessibility see https://github.com/GCTC-NTGC/gc-digital-talent/issues/5487

Comment (esizer) [reply apps/web/src/components/SkillDialog/SkillDialog.tsx:1]: Good catch, thanks! Added in [update null state to indicate family as well](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7445/commits/9a776cdf6f6a46213b3abd250536d841ad84824d)

Comment (vd1992) [reply apps/web/src/components/SkillDialog/SkillDialog.tsx:1]: Makes sense to me

File: apps/web/src/components/SkillDialog/SkillDialog.stories.tsx (hunk 2 above)
Comment (vd1992) [reply apps/web/src/components/SkillDialog/SkillDialog.stories.tsx:55]: Ah, now I get it, it adds an option to the dropdown

**Review by vd1992 — APPROVED** [review]
Looking good 😎

**Pull request merged**